src/cache/file_cache.cc

```diff
@@ -114,34 +114,101 @@ bool FileCache::Run(ui64 clean_period) {
 using namespace string;
 
 // static
-HandledHash FileCache::Hash(HandledSource code, CommandLine command_line,
-                            Version version) {
+HandledHash
+FileCache::Hash(
+  HandledSource code,
+  CommandLine command_line,
+  Version version,
+  const std::vector<string::HandledSource>& additional_sources)
```

Comment:
It's better for interface to a single `HandleSource` argument, like: `Vector<HandledSource> sources` - replacing the `code`. The 1-element vector is easy to create with braced-list constructor.